### ISSUE TITLE
iso2 missing header?

### ISSUE THREAD

**Issue by cienporcien** [EVerest/cbexigen#64]
iso2 seems to be missing the header part of the Req/Res definitions? Or maybe I'm missing something...

Here is the iso20 SessionSetupReq:

// Element: definition=complex; name={urn:iso:std:iso:15118:-20:CommonMessages}SessionSetupReq; type={urn:iso:std:iso:15118:-20:CommonMessages}SessionSetupReqType; base type=V2GRequestType; content type=ELEMENT-ONLY;
//          abstract=False; final=False; derivation=extension;
// Particle: Header, MessageHeaderType (1, 1); EVCCID, identifierType (1, 1);
struct iso20_SessionSetupReqType {
    // Header, MessageHeaderType
    struct iso20_MessageHeaderType Header;
    // EVCCID, identifierType (base: string)
    struct {
        char characters[iso20_EVCCID_CHARACTER_SIZE];
        uint16_t charactersLen;
    } EVCCID;

};

Here is the iso2 SessionSetupReq:

// Element: definition=complex; name={urn:iso:15118:2:2013:MsgBody}SessionSetupReq; type={urn:iso:15118:2:2013:MsgBody}SessionSetupReqType; base type=BodyBaseType; content type=ELEMENT-ONLY;
//          abstract=False; final=False; derivation=extension;
// Particle: EVCCID, evccIDType (1, 1);
struct iso2_SessionSetupReqType {
    // EVCCID, evccIDType (base: hexBinary)
    struct {
        uint8_t bytes[iso2_evccIDType_BYTES_SIZE];
        uint16_t bytesLen;
    } EVCCID;

};

Roger

**Comment by barsnick** [comment]
In ISO -2, `SessionSetupReq` and the other message types are elements of a top-level `V2G_Message` type, which contains a generic `Header` (of `MessageHeaderType`), and a `Body`, which contains e.g. `SessionSetupReq`. So the header is _co-located_ with the body.

In ISO -20, `SessionSetupReq` is derived from a `V2GRequestType`, in turn derived from `V2GMessageType`, which contains a `Header`. Therefore `SessionSetupReq` itself _contains_ a `Header`.

In summary, you need to encode or decode a different element type `V2G_Message` for ISO -2 communication, and you will find your `Header` there.

**Comment by cienporcien** [comment]
Ok, that makes sense. Thanks.
